Desktop_Interface/isobuffer.cpp

```diff
@@ -2,391 +2,364 @@
+	}
+	return voltageLevel;
 }
 
-
-
-short isoBuffer::inverseSampleConvert(double voltageLevel, int TOP, bool AC){
-
-    double scope_gain = (double)(virtualParent->driver->scopeGain);
-    short sample;
-
-    if(AC){
-        voltageLevel += virtualParent->currentVmean; //This is old (1 frame in past) value and might not be good for signals with large variations in DC level (although the cap should filter that anyway)??
-    }
+short isoBuffer::inverseSampleConvert(double voltageLevel, int TOP, bool AC) const
+{
+	double scope_gain = (double)(virtualParent->driver->scopeGain);
+	short sample;
+
+	if (AC)
+	{
+		// This is old (1 frame in past) value and might not be good for signals with
+		// large variations in DC level (although the cap should filter that anyway)??
+		voltageLevel += virtualParent->currentVmean;
+	}
 #ifdef INVERT_MM
-    if(virtualParent->driver->deviceMode == 7) voltageLevel *= -1;
+	if (virtualParent->driver->deviceMode == 7) voltageLevel *= -1;
 #endif
-    if (virtualParent->driver->deviceMode != 7) voltageLevel -= voltage_ref;
+	if (virtualParent->driver->deviceMode != 7) voltageLevel -= voltage_ref;
 
-    //voltageLevel = (sample * (vcc/2)) / (frontendGain*scope_gain*TOP);
-    sample = (voltageLevel * (frontendGain*scope_gain*TOP))/(vcc/2);
-    return sample;
+	//voltageLevel = (sample * (vcc/2)) / (frontendGain*scope_gain*TOP);
+	sample = (voltageLevel * (frontendGain*scope_gain*TOP))/(vcc/2);
+	return sample;
 }
 
 #define NUM_SAMPLES_SEEKING_CAP (20)
 
 #ifdef INVERT_MM
-    #define X0_COMPARISON_CAP >
-    #define X1_X2_COMPARISON_CAP <
+constexpr auto X0_COMP_FTOR = std::greater<int> {};
+constexpr auto X1_X2_COMP_FTOR = std::less<int> {};
 #else
-    #define X0_COMPARISON_CAP <
-    #define X1_X2_COMPARISON_CAP >
+constexpr auto X0_COMP_FTOR = std::less<int> {};
+constexpr auto X1_X2_COMP_FTOR = std::greater<int> {};
 #endif
 
-//For capacitance measurement.  x0, x1 and x2 are all various time points used to find the RC coefficient.
-int isoBuffer::cap_x0fromLast(double seconds, double vbot){
-    int samplesInPast = seconds * samplesPerSecond;
-    if(back < samplesInPast){
-        return -1; //too hard, not really important
-    }
-    short vbot_s = inverseSampleConvert(vbot, 2048, 0);
-    qDebug() << "vbot_s (x0) = " << vbot_s;
-
-    int num_found = 0;
-    for(int i=samplesInPast; i; i--){
-        short currentSample = buffer[back - i];
-        if(currentSample X0_COMPARISON_CAP vbot_s){
-            num_found++;
-        } else num_found--;
-        if(num_found < 0){
-            num_found = 0;
-        }
-        if (num_found > NUM_SAMPLES_SEEKING_CAP){
-            return samplesInPast-i;
-        }
-    }
-    return -1;
-}
+int capSample(isoBuffer const& self, int rbegin, int target, double seconds, double value, auto comp)
```

Comment:
Why do you need the const ref to yourself?

Reply:
(That being said though, just read the function and damn is this one more efficient)

Reply:
That's because i made that function be a free function instead of a private method.

No, reason, really. It was just late at night and made a weird decision.
It'll be a private member function on next push.